mindmeld/active_learning/data_loading.py

```diff
@@ -0,0 +1,425 @@
+            test_queries=QueryLoader(app_path, test_pattern, load, save).queries,
+            unsampled_queries=unsampled_queries,
+            sampled_queries=sampled_queries,
+            newly_sampled_queries=newly_sampled_queries,
+        )
+
+    @staticmethod
+    def get_data_bucket_for_selection(
+        app_path,
+        load,
+        save,
+        train_pattern,
+        test_pattern,
+        unlabeled_logs_path,
+        labeled_logs_pattern=None,
+        log_usage_pct=1.0,
+    ):
+        """Creates a DataBucket to be used for log selection.
+
+        Args:
+            app_path (str): Path to MindMeld application
+            load (bool): Whether to load pickled queries from a local folder
+            save (bool): Whether to save queries as a local pickle file
+            train_pattern (str): Regex pattern to match train files. For example, ".*train.*.txt"
+            test_pattern (str): Regex pattern to match test files. For example, ".*test.*.txt"
+            unlabeled_logs_path (str): Path a logs text file with unlabeled queries
+            labeled_logs_pattern (str): Pattern to obtain logs already labeled within a MindMeld app
+            log_usage_pct (float): Percentage of the log data to use for selection
+
+        Returns:
+            selection_data_bucket (DataBucket): DataBucket for log selection
+        """
+        log_queries = (
+            QueryLoader(app_path, labeled_logs_pattern, load, save).queries
+            if labeled_logs_pattern
+            else LogQueriesLoader(app_path, unlabeled_logs_path).queries
+        )
+        if log_usage_pct < 1.0:
+            sample_size = int(log_usage_pct * len(log_queries))
+            log_queries, _, _, _ = StrategicRandomSampling(sample_size).sample(
+                unsampled=log_queries
+            )
+        return DataBucket(
+            label_map=LabelMap.get_label_map(app_path, train_pattern),
+            test_queries=QueryLoader(app_path, test_pattern, load, save).queries,
+            unsampled_queries=log_queries,
+            sampled_queries=QueryLoader(app_path, train_pattern, load, save).queries,
+        )
+
+
+class DataBucket:
+    """Class to hold data throughout the Active Learning training pipeline.
+    Responsible for data conversion, filtration, and storage.
+    """
+
+    def __init__(
+        self,
+        label_map,
+        test_queries,
+        unsampled_queries,
+        sampled_queries,
+        newly_sampled_queries=None,
+    ):
+
+        self.label_map = label_map
+        self.test_queries = test_queries
+        self.unsampled_queries = unsampled_queries
+        self.sampled_queries = sampled_queries
+        self.newly_sampled_queries = (
+            newly_sampled_queries
+            if newly_sampled_queries
+            else deepcopy(sampled_queries)
+        )
+
+    def convert_queries_to_tuples(self, queries: List, class_type: str = "domain"):
```

Comment:
I dont see any code calling this function

Reply:
Good catch, removed this function.